internal/providers/virtualbox/virtualbox.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// The virtualbox provider fetches the configuration from raw data on a partition
+// with the GUID 99570a8a-f826-4eb0-ba4e-9dd72d55ea13
+
+package virtualbox
+
+import (
+	"bytes"
+	"io/ioutil"
+	"time"
+
+	"github.com/coreos/ignition/config"
+	"github.com/coreos/ignition/config/types"
+	"github.com/coreos/ignition/config/validate/report"
+	"github.com/coreos/ignition/internal/log"
+	"github.com/coreos/ignition/internal/providers/util"
+	"github.com/coreos/ignition/internal/resource"
+)
+
+const (
+	configPath = "/dev/disk/by-partuuid/99570a8a-f826-4eb0-ba4e-9dd72d55ea13"
```

Comment:
I'm not sure if any other implementations are doing this, but we are doing this due to simplicity. It allows us to has a simple, os-independent ruby script (for vagrant) that just generates a gpt partioned drive with raw data in the first (and only) partition.